### ISSUE TITLE
Mark as Spam / Flag to Leader

### ISSUE THREAD

**Issue by jterps08** [PowerlineApp/powerline-mobile#178]
We need to provide the user with the ability to mark any user post or user petition as spam (in public communities) or flag for leader review (in groups). This can be the same flag, just different UI display depending on whether it's public communities or groups. 

For **groups**, group leaders would need to able to 1) view all flagged posts (name, post, group), 1b) quickly see the five most recent posts from that user in the group, 2) delete a reported post (only if 4 or more people have flagged it), and 3) remove the user with option to ban from group (blocks user from rejoining the group). 3b) Group leaders need to be able to see a "banned list" for their group and remove users from that list if necessary. 

For **public communities,** a superuser / 'operator panel' as Austin would say would let us 1) view all flagged posts (name, post, group) in general, 1b) quickly see the five most recent posts from that user, 2) delete the post, and 3) remove the user with option to ban (ban adds user's e-mail to blacklist for registration). Ideally, we have a multi-select checkbox as part of the UI so that we can select multiple items and delete en masse if needed.

Everything above for both groups and public communities is in scope except 1b and 3b... We can do 1b and 3b separately. If possible, can we prevent a post from getting automatically boosted? For example, a user reports a post. The leader can inhibit automatic boosting for that particular post without removing the post. Is that relatively easy? If so, let's include that as item 4 for groups.

**Comment by petervojtek** [comment]
there does not seem to exist a backend endpoint for this use case...

**Comment by jterps08** [comment]
Correct. P2 item. Igor will work on this after he has answered any questions you may have on any items, P1/P2 bugs are done, P1 enhancements are done, (continues to answer questions while you start working on the web app), etc... We won't update frontend with this functionality until after web app work is done (sometime in early October).

**Comment by jterps08** [comment]
Igor - If you have no other work on P1/P2 bugs to do or anything that Peter needs from you, please start on this.

**Comment by igaponov** [comment]
Get posts with a filtration: https://api-dev.powerli.ne/api-doc#get--api-v2-groups-{group}-posts
Get petitions with a filtration: https://api-dev.powerli.ne/api-doc#get--api-v2-groups-{group}-user-petitions
Update user status in a group: https://api-dev.powerli.ne/api-doc#patch--api-v2-groups-{group}-users-{user}
Get group's users with a filtration: https://api-dev.powerli.ne/api-doc#get--api-v2-groups-{id}-users

@jterps08 you can test admin panel as well, look for post/petition menu item at the left.

**Comment by jterps08** [comment]
@igaponov What are the endpoints for marking an item as spam or flagging it to leader?

**Comment by jterps08** [comment]
I also think that we are going to need an endpoint that allows us to count the number of flagged items in a particular group. Also, can a leader review an item and ignore the flag so that it is removed from the flagged-list? 

e.g Save the Whales has 3 flagged items pending review... Save the Trees had 10 items but the leader ignored all of them so Save the Trees has 0 flagged items pending review... United States of America has 20 flagged items pending review (that will be fun). 

I will want a screen that a group manager can go to and see all the items that have been flagged in his group. I will want to see how many times each item was reported. And 3 buttons that will enable the leader to ignore, delete, and remove the user from the group. Tapping Ignore will remove the item from the list until it is reported again. Tapping delete will remove the item from the group. Tapping Remove user will remove the user from the group. I will create a mockup and share below.

**Comment by jterps08** [comment]
![image](https://cloud.githubusercontent.com/assets/7582457/23380886/7f522244-fd0a-11e6-8dff-db9610877555.png)

**Comment by devotebest** [comment]
should I work now?

**Comment by igaponov** [comment]
To mark as a spam/mark as not a spam:
`POST|DELETE /api/v2/posts/{id}/spam`
`POST|DELETE /api/v2/user-petitions/{id}/spam`

**Comment by jterps08** [comment]
Thanks @igaponov ... this is ready for frontend work now I think.
